### PR TITLE
bump aws-cdk to 1.153.1

### PR DESCRIPTION
**Pull request by steved**
Adds support for K8s 1.22.

### CODE REVIEW

**Review by Secretions — APPROVED** [review]
Should also update the bootstrap/readme

**Comment by steved** [comment]
Pending testing / resolution of https://github.com/aws/aws-cdk/pull/20000 🙃

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -121,6 +121,10 @@ jobs:
         LOG_DIR: k8s-cluster-state
       run: |
         docker run --rm -v $(pwd):/cdk -v $(pwd)/agent_logs:/domino-deployer/logs -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY -e KUBECONFIG=/cdk/kubeconfig $DEPLOYER_IMAGE python -m fleetcommand_agent run -f /cdk/domino.yml
+    - name: install-kubectl
+      uses: azure/setup-kubectl@v2.0
+      with:
+        version: 'v1.21.11'
```
Comment (steved) [inline .github/workflows/tests.yml:127]: needed to workaround the EKS client auth errors with 1.24

File: cdk/domino_cdk/aws_configurator.py [hunk 2]
```diff
@@ -45,17 +53,30 @@ def install_calico(self):
...
-                "calico-crds",
-                cluster=self.eks_cluster,
-                manifest=[crd for crd in loaded_manifests if crd["kind"] == "CustomResourceDefinition"],
+            crd_manifests += [crd for crd in loaded_manifests if crd["kind"] == "CustomResourceDefinition"]
+            notcrd_manifests += [notcrd for notcrd in loaded_manifests if notcrd["kind"] != "CustomResourceDefinition"]
+
+        # NB: be careful changing resource names or removing the manifests as prune is set by default
+        # and could inadvertantly remove other resources if deleted after the new manifest is created
+
+        # Split CRDs into multiple manifests so they don't go over the lambda limit of 262144 bytes
```
Comment (Secretions) [inline cdk/domino_cdk/aws_configurator.py:62]: Does this have any noticeable impact on install time?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -121,6 +121,10 @@ jobs:
         LOG_DIR: k8s-cluster-state
       run: |
         docker run --rm -v $(pwd):/cdk -v $(pwd)/agent_logs:/domino-deployer/logs -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY -e KUBECONFIG=/cdk/kubeconfig $DEPLOYER_IMAGE python -m fleetcommand_agent run -f /cdk/domino.yml
+    - name: install-kubectl
+      uses: azure/setup-kubectl@v2.0
```
Comment (Secretions) [inline .github/workflows/tests.yml:125]: Gerardo pinned kubectl in the past day or so, so you should probably rebase. I don't have an opinion on which install method to go with, but we'll be doing it twice if this merges.

File: cdk/domino_cdk/aws_configurator.py (hunk 2 above)
Comment (steved) [reply cdk/domino_cdk/aws_configurator.py:62]: ~30min for your last commit in master, ~31min for my last commit here

**Pull request merged**